pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -169,12 +170,15 @@ def restore_training_state(self) -> None:
         if not self._loaded_checkpoint:
             return
 
-        # restore precision plugin (scaler etc.)
-        self.trainer.precision_plugin.on_load_checkpoint(self._loaded_checkpoint)
         # restore loops and their progress
         self.restore_loops()
 
-        self.restore_optimizers_and_schedulers()
+        if self.trainer.state.fn == TrainerFn.FITTING:
+            # restore precision plugin (scaler etc.)
+            self.trainer.precision_plugin.on_load_checkpoint(self._loaded_checkpoint)
```

Comment:
Isn't it possible that we want to load this state for evaluation? precision can be enabled for them

Reply:
hmm. good point. done. thanks